distributed/tests/test_worker.py

```diff
@@ -1251,21 +1264,33 @@ def test_startup2():
     yield worker._close()
 
     # Register another preload function
-    response = yield c.register_worker_callbacks(setup=mystartup2)
-    assert len(response) == 2
+    names2 = yield c.register_worker_callbacks(setup=mystartup2)
+    assert len(names2) == 1
     assert len(s.worker_setups) == 2
 
     # Check it has been run
     result = yield c.run(test_startup2)
     assert list(result.values()) == [True] * 2
 
+    # unregister a preload function
```

Comment:
I changed 'register_worker_callbacks' to return the registered names to allow unregisterring callback functions with inferred names.